pulp_python/app/tasks/sync.py

```diff
@@ -1,6 +1,8 @@
 import logging
 import tempfile
+from typing import Optional, Any, AsyncGenerator
```

Comment:
This turned out to become a rather hot line...
I adopted the version `import typing as t` with positive experience.